src/mscorlib/src/System/String.cs

```diff
@@ -411,7 +415,61 @@ private unsafe static bool EqualsHelper(String strA, String strB)
                 return (length <= 0);
             }
         }
-        
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        private unsafe static bool OrdinalCompareSubstring(String strA, String strB, int length)
+        {
+            Contract.Requires(strA != null);
+            Contract.Requires(strB != null);
+            Contract.Requires(strA.Length <= length);
+            Contract.Requires(strB.Length <= length);
+
+            fixed (char* ap = &strA.m_firstChar) fixed (char* bp = &strB.m_firstChar)
+            {
+                char* a = ap;
+                char* b = bp;
+
+#if AMD64
+#if ALIGN_ACCESS
+    #error Needs porting for ALIGN_ACCESS (https://github.com/dotnet/corert/issues/430) 
+#endif // ALIGN_ACCESS
```

Comment:
I think the fast path for the first char in the caller is good.

The comparison that I am suggesting wound is not completely redundant with the check in the caller because of it would compare the whole int32 (two characters).